org.hl7.fhir.publisher.core/src/main/java/org/hl7/fhir/igtools/publisher/TranslationImplementation.java

```diff
@@ -104,5 +120,38 @@ public String translateAndFormat(String contest, String lang, String string2, Ob
     return null;
   }
 
+  public void printKeys(){
+    if(harvest) {
+      System.out.println("Printing bundle keys:");
+      keys.forEach((key, value) -> System.out.println(String.format("%s=%s", key, value)));
+    } else {
+      System.out.println("Not harvesting bundle keys. Nothing printed");
+    }
+  }
+
+  private String generateKey(String context, String path) {
+    String prefix = StringUtils.isNotBlank(context) ? context + "." : "";
+    String key = prefix + path.toLowerCase()
+            .replaceAll(nonAlpha, "_")
+            .replaceAll(noEndDash, "");
+
+    if(harvest) {
+      if(keys.containsKey(key) && !path.equalsIgnoreCase(keys.get(key))) {
+        System.out.println(String.format("Overwriting key [%s]. Old value [%s]. New value [%s].", key, keys.get(key), path));
+      }
+      keys.put(key, path);
+    } else {
+      if(!bundle.containsKey(key)) {
+        System.out.println(String.format("Message key not found : [%s]", key));
+      }
+    }
+
+    return key;
+  }
 
+  // TODO: Handle by custom ResourceBundle.Control instead. This is just a workaround
+  // Should language properties be partitioned into bundles by context instead of current prefixing?
+  private String resolve(String key) {
+    return new String(bundle.getString(key).getBytes(StandardCharsets.ISO_8859_1), StandardCharsets.UTF_8);
+  }
 }
```

Comment:
I know that testing for this library is where it needs to be currently, but I would prefer if you added tests to cover the new functionality in this file you have added. Even if it's just testing the key management method and the calls the the modified methods that now check if "harvest" is set. I'm working on getting some better CICD pipelines set up for a bunch of the core libraries, including this one and this is the kind of approach that would be good to take moving forward.

If not too complicated, it would be great to test the actual fetching of pairs using the 'harvest' variable.